src/header/common/referer.rs

```diff
@@ -0,0 +1,35 @@
+use header::{Header, HeaderFormat};
+use std::fmt::{self, Show};
+use header::shared::util::from_one_raw_str;
+
+/// The `Referer` header.
+///
+/// The Referer header is used by user agents to inform server about
+/// the page URL user has came from.
+///
+/// See alse [RFC 1945, section 10.13](http://tools.ietf.org/html/rfc1945#section-10.13).
+///
+/// Currently just a string, but maybe better replace it with url::Url or something like it.
```

Comment:
This is the case, which makes me want for Rust to support implicits, like Scala.

Anyway, I wouldn't like to make special exception for base URL in respect to passing to around to header parsers, as it looks not generic enough for me.